tests/python/pants_test/backend/python/rules/test_inject_init.py

```diff
@@ -0,0 +1,39 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from pants.backend.python.rules.inject_init import InitInjectedDigest
+from pants.engine.fs import EMPTY_DIRECTORY_DIGEST, EMPTY_SNAPSHOT, Snapshot
+from pants.engine.rules import RootRule
+from pants.util.collections import assert_single_element
+from pants_test.test_base import TestBase
+
+
+class TestInjectInit(TestBase):
+
+  @classmethod
+  def rules(cls):
+    return super(TestInjectInit, cls).rules() + [RootRule(Snapshot)]
+
+  def assert_result(self, input_snapshot, expected_digest):
+    injected_digest = assert_single_element(
```

Comment:
This `assert_single_element` call can be replaced by a single comma. Just saying! =D

Reply:
Hah true. I'm personally rooting for us to add `scheduler.single_product_request()` and converge on that.

Reply:
And @Eric-Arellano I would really like to encourage task parallelism if possible (requesting multiple things at once) and using tuples by default encourages that with a really neat syntax!

Reply:
>  using tuples by default encourages that with a really neat syntax!

To clarify, this means that I _should_ use the tuple syntax of `injected_digest, = blah()`, right?

Why is this the case that using tuple syntax would increase parallelism? Is the idea that it's a good practice, or will it actually result in that here?

Reply:
I'm happy with this as-is.

I agree with some other folks that the `foo, = bar` syntax is pretty subtle and can be confusing, and we should probably avoid it.

In test code, I'd advocate for introducing:
```python
class TestBase
  def single_scheduler_product_request(self, product, subject):
    return assert_single_element(self.scheduler.product_request(product, [subject]))
```

which IIRC we used to have before, but I can't find?

I'm mildly against `scheduler.single_product_request` as it may discourage parallelism in non-test code.

Reply:
To add to the noise, I also find the `foo, = bar` syntax very confusing and easy to miss, the seconds we save when typing longer accessors are all lost when we spend an extra hour or two going on fruitless debug tangents when we miss the comma and assume things that are not true. Rant over.

I'm in favour of the helper method being in `TestBase` vs the `scheduler`, since I agree that we should make it slightly painful to request only one product. This may avoid problems like [this](https://github.com/pantsbuild/pants/pull/6577) in the future (not v2-specific, but it's the same UX reasoning).

Reply:
> I'm mildly against scheduler.single_product_request as it may discourage parallelism in non-test code.

Had no idea that'd be a thing. Good to know, and agreed this can live in `TestBase`. I can put up a PR later today to add this and refactor every current usage of the alternatives.

Reply:
> I also find the `foo, = bar` syntax very confusing and easy to miss

Ok, I hadn't realized this. It might make sense to have an explicit single product request method if we make it clear in the docstring how to request in parallel (since tuple destructuring with more than one element should be less ambiguous, I think?). Not sure.

*But also, syntax highlighting saves lives!*

> the seconds we save when typing longer accessors

If anyone isn't using an editor that allows them to autocomplete python symbols, we could consider adding documentation on how to do that in their preferred editor perhaps!